eth/downloader/receiptreference.go

```diff
@@ -0,0 +1,125 @@
+package downloader
+
+import (
+	"bytes"
+	"embed"
+	"encoding/gob"
+	"path"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// userDepositNonces is a struct to hold the reference data for user deposits
+// The reference data is used to correct the deposit nonce in the receipts
+type userDepositNonces struct {
+	ChainID uint64 `json:"chainId"`
+	// "Start" and "End" would be more appropriate names. Last is not inclusive.
```

Comment:
Should just do this unless it messes with the encoding?

Reply:
It does indeed mess with the encoding. The receipt reference tool mostly uses the words "Start" and "End", but the underlying data structure was/is still using "First" and "Last". Changing the structure here makes the data fail to decode.